### ISSUE TITLE
Timeout while uploading large files in GeoNode/GeoServer

### ISSUE THREAD

**Issue by satigebah** [GeoNode/geonode#7871]
Hello I have some problems with geonode in docker.
When I try to download a large layer, I have this problem:

**Unexpected error! - 504 Gateway Timeout. GeoNode lost the connection with GeoServer or DB due to a connection timeout. Consider using the management commands to import data!**

**Comment by t-book** [comment]
The title says uploading,
the description downloading.
which one is it?

- have a look at geoservers memory usage
- have a look at uWSGI harakiri which lives in your django / celery container
    - https://github.com/GeoNode/geonode/blob/3.2.x/uwsgi.ini#L24

**Comment by zahirulislamovich** [comment]
basically i am facing uploading issue... it will be a pleasure having clear instructions of changing file upload size and increase time. its bit confusing and stlll unresolved

**Comment by t-book** [comment]
@zahirulislamovich Unfortunately transferring a large amount over HTTP is cumbersome. For example, you can face many bottlenecks like: nginx max body size limits, python worker timeouts, geoserver memory issues, and so on. Is the management command importlayers of help for your task? Using it you can workaround several possible problems

**Comment by zahirulislamovich** [comment]
@t-book thanks. it is very useful as i currently use this to upload large files. (docker manage.py updatelayers) , @satigebah  i assume this is the only solution as i am facing this sorts of issue for last 2 years which was solved only by manage layer. i appreciate geonode team for their great task.  it will be the best of having a tutorial to external geoserver_data integration (for large files) and gdal integration. if its possible to ingest .sid file we can store so many gb data; as xample i have 25gb orthophoto in tiff but in .sid foormat its just 6 gb. Thanks again tony

**Comment by satigebah** [comment]
@t-book @zahirulislamovich  Thank you for everything! I tested it and was able to add the layers with python manage.py importlayers

**Comment by iandmozart** [comment]
I had the same issue a few days ago.
My instance was setup following the Geonode Project on Docker instructions (https://docs.geonode.org/en/master/install/advanced/project/index.html)

What I did was based on t-follow comment above, and this link: https://gis.stackexchange.com/questions/382710/timeout-while-uploading-large-files-in-geonode-geoserver

Steps:
- Go to your project folder, check the uswgi.ini file:
cd /opt/geonode_custom/my_geonode
nano uwsgi.ini

- Check the line when it says: harakiri = 60 --> increase this parameter to 600:
harakiri = 600

- Save and exit
- Restart the containers:
docker-compose stop
docker-compose up -d

**Comment by EHJ-52n** [comment]
I had the same issue, which is reported with the following error message:

```
Unexpected error! - Unknwon Error Code.
Contact the system administrator for more information regarding this error message.
Additional info:
    [Server Error: timeout
     Please check your network connection.
     In case of Layer Upload make sure GeoServer is running and accepting connections.]
```

To solve this, I increased the parameter `harakiri` to 600  as outlined above.
In addition, I added the recommendation to upload larger files or when having a bad connection, using a backend driven approach:
* upload to geonode container + importlayers management command
* upload to geoserver + updatelayers management command.

Hope this helps, because the error message was quite confusing for me.

Maybe, someone can improve the error message or the data loading documentation, if not already done during the latest developments.
